Assets/Scripts/UpgradeStore.cs

```diff
@@ -25,10 +40,12 @@ void OnDestroy()
     }
 
 
-    public void OnUpgrade(Upgrade msg)
+    public void OnUpgrade(Upgrade msg, Price msg2)
```

Comment:
Can't have an event handler with 2 arguments like this in Code Control. 
You can have several fields in a class tho.